### PR TITLE
Process re-ordering and updated logic for skip* steps

### PR DESCRIPTION
**Pull request by drpatelh**
# nf-core/viralrecon pull request

Many thanks for contributing to nf-core/viralrecon!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/viralrecon branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/viralrecon)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/viralrecon/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by drpatelh** [comment]
Tests will fail again unfortunately. Sorry 😞 Need to sort out a much smaller Kraken2 database that isnt human...

File: docs/usage.md [hunk 1]
```diff
@@ -14,20 +14,17 @@
...
-  * [`--host_kraken2_db`](#--host_kraken2_db)
-  * [`--host_kraken2_name`](#--host_kraken2_name)
-  * [`--viral_kraken2_db`](#--viral_kraken2_db)
-  * [`--viral_kraken2_name`](#--viral_kraken2_name)
-  * [`--kraken2_use_host_db`](#--kraken2_use_host_db)
+  * [`--kraken2_db`](#--kraken2_db)
+  * [`--kraken2_db_name`](#--kraken2_db_name)
   * [`--kraken2_use_ftp`](#--kraken2_use_ftp)
-  * [`--save_kraken2_fastq`](#--save_kraken2_fastq)
+  * [`--save_kraken2_fastq`](#--save_kraken2_fastq)
```
Comment (heuermh) [inline docs/usage.md:25]: A little trailing whitespace there 😉

File: main.nf [hunk 2]
```diff
@@ -18,40 +18,32 @@ def helpMessage() {
...
     Kraken2
-      --host_kraken2_db [file]        Full path to Kraken2 database built from host genome
-      --host_kraken2_name [str]       Name of host genome for building Kraken2 database e.g. 'human'
-
-      --viral_kraken2_db [file]       Full path to Kraken2 database built from viral genome
-      --viral_kraken2_name [str]      Name of viral genome for building Kraken2 database e.g. 'viral'
-
-      --kraken2_use_host_db [bool]    Use Kraken2 host database instead of viral one for filtering reads
+      --kraken2_db [file]             Full path to Kraken2 database built from host genome (Default: kraken2_human.tar.gz hosted on Zenodo)
+      --kraken2_db_name [str]         Name of host genome for building Kraken2 database (Default: 'human')
```
Comment (heuermh) [inline main.nf:41]: Much better!

File: main.nf [hunk 3]
```diff
@@ -1197,104 +927,199 @@ process BCFTOOLS_CONSENSUS {
...
+    }
+} else {
+    exit 1, "NXF_OFFLINE=true or -offline has been set so cannot download files required to build Kraken2 database!"
+}
+
+/*
+ * STEP 4.1: Filter reads with Kraken2
+ */
+// TODO nf-core: Add Kraken2 log output to MultiQC
+// TODO nf-core: Add shell script to trim file extensions and return appropriate file name
```
Comment (heuermh) [inline main.nf:1025]: Would `simpleName` from Nextflow work for that?

**Pull request merged**